public/js/graph/react_node.js.jsx

```diff
@@ -203,60 +299,165 @@ var ReactNodes = React.createClass({
+                currentEdge.setState({status: currentEdge.state.status}, 
+                                     function(){this.updateEdge(svg)}.bind(currentEdge));
+            });
+        }
+
+    },
+    
+    turn: function(svg){
+        if (this.isSelected()) {  
+            if (this.props.parents.length == 0){
+                this.setState({status: 'takeable'}, function(){this.updateNode(svg)}.bind(this));
+            } else {
+                this.setState({status: 'inactive'}, function(){this.updateNode(svg)}.bind(this));
+            }
+        } else {
+            this.setState({status: 'active'}, function(){this.updateNode(svg)}.bind(this));
+        }
+                        
+        $.each(this.props.childs, function (i, node) {
+            var currentNode = svg.refs['nodes'].refs[node] ? svg.refs['nodes'].refs[node] :
+                                                             svg.refs['bools'].refs[node];
+            currentNode.setState({status: currentNode.state.status},
+                                 function(){this.updateNode(svg)}.bind(currentNode));
+        });
+        
+        $.each(this.props.outEdges, function (i, edge) {
+            var currentEdge = svg.refs['edges'].refs[edge];
+            currentEdge.setState({status: currentEdge.state.status}, 
+                                 function(){this.updateEdge(svg)}.bind(currentEdge));
+        });
+        
+        $.each(this.props.inEdges, function (i, edge) {
+            var currentEdge = svg.refs['edges'].refs[edge];
+            currentEdge.setState({status: currentEdge.state.status}, 
+                                 function(){this.updateEdge(svg)}.bind(currentEdge));
+        });
+    },
+    
+    focusPrereqs: function(svg){
+        if (this.state.status !== 'active') {
+            if (this.state.status !== 'overridden') {
+                this.setState({status: 'missing'});
+            }
+            
+            $.each(this.props.inEdges, function (i, edge) {
+                var currentEdge = svg.refs['edges'].refs[edge];
+                var sourceNode = svg.refs['nodes'].refs[currentEdge.props.source] ? svg.refs['nodes'].refs[currentEdge.props.source] :
+                                                                                    svg.refs['bools'].refs[currentEdge.props.source];
+                if (sourceNode.state.status !== 'active') {
+                    currentEdge.setState({status: 'missing'});
+                }
+            });
+            
+            $.each(this.props.parents, function (i, node) {
+                var currentNode = svg.refs['nodes'].refs[node] ? svg.refs['nodes'].refs[node] :
+                                                                 svg.refs['bools'].refs[node];
+                currentNode.focusPrereqs(svg);
+            });
+        }
+    },
+    
+    unfocusPrereqs: function(svg){
+        if (!this.isSelected()) {
+            /*if (activeFocus === '' ||
```

Comment:
I don't really know what this code is for, but if you think you can remove it, please do so.